sdks/python/apache_beam/runners/portability/fn_api_runner.py

```diff
@@ -1122,9 +1124,25 @@ def restore(self):
     def process_instruction_id(self, unused_instruction_id):
       yield
 
-    def blocking_get(self, state_key):
+    def blocking_get(self, state_key, continuation_token=None):
       with self._lock:
-        return b''.join(self._state[self._to_key(state_key)])
+        full_state = self._state[self._to_key(state_key)]
+        if self._use_continuation_tokens:
+          new_token = 'token_%d' % len(self._continuations)
+          if not continuation_token:
+            # Store (index, blobs).
+            self._continuations[new_token] = 0, tuple(full_state)
+            return b'', new_token
+          else:
+            ix, full_state = self._continuations[continuation_token]
+            if ix == len(full_state):
+              return b'', None
+            else:
+              self._continuations[new_token] = ix + 1, full_state
+              return full_state[ix], new_token
```

Comment:
So every time there's a valid continuation token and the iteration is advanced, that's stored in a new entry in the list of continuations. Is there any chance that this list may have get identical continuations with different tokens? Since this list seems to be only added to, I instinctively worry about it leaking memory like that. Or is it a non-issue based on the broader architecture?